From: Carlos Garnacho <carlosg@gnome.org>
Date: Wed, 16 Dec 2020 19:14:54 +0100
Subject: backend/native: Check libinput_device in async call

Power saving changes in laptop panels enable/disable the attached
touchscreen input device, this is an asynchronous operation that
may be happening while the device is disappearing.

In fact, closing the lid is such perfect storm where both things
happen around the same time.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1644>
---
 src/backends/native/meta-input-settings-native.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/backends/native/meta-input-settings-native.c b/src/backends/native/meta-input-settings-native.c
index 39cd5f6..d0c3188 100644
--- a/src/backends/native/meta-input-settings-native.c
+++ b/src/backends/native/meta-input-settings-native.c
@@ -108,7 +108,8 @@ set_send_events (GTask *task)
     }
 
   libinput_device = meta_input_device_native_get_libinput_device (device);
-  libinput_device_config_send_events_set_mode (libinput_device, libinput_mode);
+  if (libinput_device)
+    libinput_device_config_send_events_set_mode (libinput_device, libinput_mode);
 
   return G_SOURCE_REMOVE;
 }
